hphp/runtime/vm/jit/vasm-instr.h

```diff
@@ -1156,10 +1154,8 @@ struct uxth { Vreg16 s; Vreg32 d; };
 /*
  * ppc64 intrinsics.
  */
-struct extrb { Vreg8 s; Vreg8 d; };   // Extract and zeros the upper bits
-struct extrw { Vreg16 s; Vreg64 d; }; // Extract and zeros the upper bits
-struct extsb { Vreg64 s; Vreg64 d; }; // Extend byte sign
-struct extsw { Vreg64 s; Vreg64 d; }; // Extend word sign
+struct extsb { Vreg8 s; Vreg64 d; };  // Extend byte sign
+struct extsw { Vreg32 s; Vreg64 d; }; // Extend word sign
```

Comment:
This should either be called `extsl{}` (extend long sign), or you should change the `Vreg32` operand to `Vreg16`.